nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBConverter.java

```diff
@@ -57,22 +64,35 @@
     /** The key to use in memory to map the database native "_id". */
     protected final String idKey;
 
+    /** The keys for booleans whose value is true or null (instead of false). */
+    protected final Set<String> trueOrNullBooleanKeys;
+
+    /** The keys whose values are ids and are stored as longs. */
+    protected final Set<String> idValuesKeys;
+
     /**
      * Constructor for a converter that does not map the MongoDB native "_id".
      *
      * @since 10.3
      */
     public MongoDBConverter() {
-        this(null);
+        this(null, Collections.emptySet(), Collections.emptySet());
     }
 
     /**
      * Constructor for a converter that also knows to optionally translate the native MongoDB "_id" into a custom id.
+     * <p>
+     * When {@code idValuesKeys} are provided, the ids are stored as longs.
      *
      * @param idKey the key to use to map the native "_id" in memory, if not {@code null}
+     * @param trueOrNullBooleanKeys the keys corresponding to boolean values that are only true or null (instead of
+     *            false)
+     * @param idValuesKeys the keys corresponding to values that are ids
      */
-    public MongoDBConverter(String idKey) {
```

Comment:
Don't we want to keep the existing constructor/methods for backward compatibility, to be safe? (I know it's mostly internal stuff)

Reply:
This is too low-level internals to make sense.